tick/optim/model/tests/generalized_linear_model.py

```diff
@@ -51,3 +51,23 @@ def run_test_for_glm(self, model, model_spars=None,
             np.testing.assert_almost_equal(model.grad(coeffs),
                                            model_spars.grad(coeffs),
                                            decimal=10)
+
+        # If the model has an intercept (ModelCoxReg does not for instance)
+        if hasattr(model, 'fit_intercept'):
+            # For the model with intercept only, test that
+            if model.fit_intercept:
+                X = model.features
+                y = model.labels
+                coeffs = np.random.randn(model.n_coeffs)
+                grad1 = model.grad(coeffs)
+
+                X_with_ones = np.hstack((X, np.ones((model.n_samples, 1))))
+                model.fit_intercept = False
+                model.fit(X_with_ones, y)
+                grad2 = model.grad(coeffs)
+
+                np.testing.assert_almost_equal(grad1, grad2, decimal=10)
+
+                # Put back model to its previous status
+                model.fit_intercept = True
+                model.fit(X, y)
```

Comment:
Can we isolate this test? The merged thing is a verrry big test. We are not true unit testers but still, we try to pretend we are :)